crucible-mir/src/Mir/PP.hs

```diff
@@ -323,7 +331,14 @@ instance Pretty ConstVal where
     pretty (ConstFunction a)   = pr_id a
     pretty (ConstInitializer a) = pr_id a
     pretty (ConstStaticRef a) = pretty "&" <> pr_id a
-    pretty ConstZST = pretty "<ZST>"
+    pretty ConstZST =
+      -- A ConstZST value represents a function definition, but because this is
```

Comment:
>A ConstZST value represents a function definition

I think this comment is overly specific.  It's currently the case that we use `ConstZST` only for `TyFnDef`, but it would be reasonable (IMO) for mir-json to use it for zero-sized structs/enums, empty tuples and closures, and zero-length arrays.  It looks like the code in `Trans.hs` that actually handles `ConstZST` actually happens to support all of these cases already.

Given that all the useful information is in the type in all of these cases, maybe it would be better to have the `Pretty Constant` instance always print the type if the expression is `ConstZST` (with a special case for `TyFnDef` if the `Pretty Ty` output doesn't look good).